src/classes/RD2_ScheduleService.cls

```diff
@@ -104,9 +100,35 @@ public without sharing class RD2_ScheduleService {
 
         newSchedules.add(schedule);
 
+        if (rd.RecurringType__c == RD2_Constants.RECURRING_TYPE_FIXED && rd.npe03__Installments__c != null) {
```

Comment:
This code and the next method can be replaced with `setFixedLengthEndDate(newSchedules, rd);`

where

```

    /***
    * @description Calculates and set the end date of Fixed Length Recurring Donation Schedule
    * @param schedules Schedules on a Recurring Donation
    * @param rd Recurring Donation record 
    * @return void
    */
    private void setFixedLengthEndDate(List<RecurringDonationSchedule__c> schedules, npe03__Recurring_Donation__c rd) {
        if (rd.RecurringType__c != RD2_Constants.RECURRING_TYPE_FIXED || rd.npe03__Installments__c == null) {
            return;
        }
        
        List<Installment> installments = getVisualizedInstallments(
            currentDate,//Is this correct? TODO
            rd.npe03__Installments__c.intValue(),
            schedules
        );

        Date endDate = installments.isEmpty() 
            ? currentDate
            : installments[installments.size() - 1].nextDonationDate;        

        for (RecurringDonationSchedule__c schedule : schedules) {
            schedule.EndDate__c = endDate;
        }
    }
```

After reviewing the code, it can be noticed that when calculating number of projected installments, we calculate number of installments from the current date, so if there an RD created on 1/1/2020 with the same date established, and Number of Planned Installments = 6 and there are 1/1/2020 and 2/1/2020 Opps, then the end date should be 6/1/2020, but it is actually 9/1/2020. The code logic to determine the last Next Donation Date for the fixed-length RD needs to be corrected.